autogl/solver/utils.py

```diff
@@ -49,64 +71,74 @@ def set_major_field(self, field) -> None:
         -------
         None
         """
-        if field in self.keys and not field == "name":
-            self.major_field = field
+        if field in self.__fields:
+            self.__major_field = field
         else:
             LOGGER.warning(
-                "do not find major field %s in current leaderboard, will ignore.", field
+                "do not find major field %s in the current LeaderBoard, will ignore.", field
             )
 
-    def insert_model_performance(self, name, performance) -> None:
+    def add_performance(
```

Comment:
The LeaderBoard is designed in dictionary format, which means: the models and rows of dataframe are 1-to-1 pairs. The names of models are **not allowed** to be the same. In your new design, at least names + representations are **not allowed** to be the same.
Please change the logic back.